### PR TITLE
PR: Add generic name to unnamed dockwidgets

### LINKED ISSUE #2578: Set objectName on all QDockWidgets to avoid warning at shutdown
## 🐛 Bug
I've been seeing this warning a lot when closing napari:

```
WARNING: QMainWindow::saveState(): 'objectName' not set for QDockWidget 0x7f8cf9467330 ';
```

the docs on [saveState](https://doc.qt.io/qt-5/qmainwindow.html#saveState) say:

> The objectName property is used to identify each QToolBar and QDockWidget. You should make sure that this property is unique for each QToolBar and QDockWidget you add to the QMainWindow

so we should probably make sure to add some sort of objectName to each of our QDockWidgets in 

https://github.com/napari/napari/blob/c618f310a5c837e77799f2d80286eea1db214f57/napari/_qt/widgets/qt_viewer_dock_widget.py#L21

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

Screen Shot 2021-04-22 at 8 36 36 PM

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

Fixes #2578

# How has this been tested?

- [x] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
looks good to me.  The only thing I'm wondering is whether someone will be surprised to see their `widget.objectName()` in the title bar, if they (for instance) use some less-than readable string for styling or who knows what.  sorry if I was confusing before, I meant that we should perhaps use `objectName` if it exists for `dock_widget.objectName()` ... but probably not for `.name` (which gets used in the `QtCustomTitleBar` a little lower)?

that said, I'd defer to you on this one.  so this is good by me

**Comment by sofroniewn** [comment]
I also don't have strong feelings here - either is an improvement on what we have now. Once tests pass, I'll be happy to merge

**Review by sofroniewn — APPROVED** [review]
Tests now passing, this approach looks good - we can change later if we run into strange behavior. Thanks!!

**Pull request merged**
